src/Common.cpp

```diff
@@ -166,10 +167,49 @@ std::vector<std::string> NAS2D::split(std::string str, char delim /*= ','*/, boo
 	ss.clear();
 
 	std::string curString{};
-    while(std::getline(ss, curString, delim)) {
-        if(skip_empty && curString.empty()) { continue; }
+	while (std::getline(ss, curString, delim))
+	{
+		if (skip_empty && curString.empty()) { continue; }
 		result.push_back(curString);
-    }
+	}
+	result.shrink_to_fit();
+	return result;
+}
+
+
+std::string NAS2D::join(std::vector<std::string> strs, char delim, bool skip_empty /*= true*/)
+{
+	auto acc_op = [](const std::size_t& a, const std::string& b) -> std::size_t { return a + static_cast<std::size_t>(1u) + b.size(); };
+	auto total_size = std::accumulate(std::begin(strs), std::end(strs), static_cast<std::size_t>(0u), acc_op);
```

Comment:
The +1 accounts for if that iteration the string is empty and only should contain the delimiter. It's intent is to accumulate a "potential maximum size" by calling reserve to increase capacity and prevent reallocations. I shrink_to_fit before returning to put it back to the actual size.

Reply:
Yes, the `+1` over `strs.size()` iterations of the loop will sum up to `strs.size()`. I'm suggesting factoring that term out. The `-1` is because there is no delimiter following the last element.

 Both ways are independent of any of the strings being empty or not.

I appreciate that you took the time to use `reserve` to prevent reallocations from affecting performance.